pkgs/stdenv/generic/make-derivation.nix

```
@@ -287,32 +352,89 @@ else let
   references = nativeBuildInputs ++ buildInputs
             ++ propagatedNativeBuildInputs ++ propagatedBuildInputs;
 
+  getSpliced = type: dep:
+    dep.__spliced or
+      (let approximate-pname = dep.pname or dep.name; in
+        assert
+          (dep?stdenv &&
+           dep.stdenv.buildPlatform != dep.stdenv.targetPlatform &&
+
+           # I can't test on Darwin, so let's just ignore it for now.
+           !dep.stdenv.buildPlatform.isDarwin &&
+           !dep.stdenv.hostPlatform.isDarwin &&
+           !dep.stdenv.targetPlatform.isDarwin &&
+
+           !(list-of-shame ? ${approximate-pname})
+          )
+
+            # Hi there!  If you're reading this, it's probably
+            # because the line below caused your PR to fail CI.
+            # Don't panic!  Most likely, what happened is that you
+            # used one of the following as a dependency:
+            #
+            # - `buildPackages.something`
+            # - `targetPackages.something`
+            # - `pkgs{Build,Host,Target}{Build,Host,Target}.something`
+            #
+            # You can't use those as dependencies (i.e. in a
+            # `buildInputs`, `nativeBuildInputs`, or
+            # `deps{Build,Host,Target}{Build,Host,Target}`
+            # attribute).  The reason is a bit obscure, but the fix
+            # is easy: just use `something` instead!  The explicit
+            # packagesets (the three bullet points above) are mainly
+            # for when you need to reference a package with string
+            # interpolation (e.g. "cat blah | ${buildPackages.jq}").
+            # For dependencies, you control which packageset is used
+            # by *which attribute you put the dependency in* -- if
+            # you put it in `depsBuildHost`, it will get pulled from
+            # `pkgsBuildHost`.
+            #
+            # Please take a moment to try to fix your PR.  If you
+            # can't get it fixed, ping @amjoseph-nixpkgs who can
```

This should be probably changed than:

```suggestion
            # can't get it fixed, ping @amjoseph-nixpkgs who can
```

---

@Mic92 it looks like the suggested change is what's currently there. The buttons to accept the change are greyed out.